packages/app/src/providers/mock/utils.ts

```diff
@@ -82,20 +81,20 @@ export function getChildrenPaths(
 }
 
 export async function cancellableDelay(
-  abortSignal?: AbortSignal,
+  abortSignal: AbortSignal,
 ): Promise<void> {
   await new Promise<void>((resolve, reject) => {
     const timeout = setTimeout(() => {
-      abortSignal?.removeEventListener('abort', handleAbort);
+      abortSignal.removeEventListener('abort', handleAbort);
       resolve();
     }, SLOW_TIMEOUT);
 
     function handleAbort() {
       clearTimeout(timeout);
-      abortSignal?.removeEventListener('abort', handleAbort);
-      reject(new AbortError(abortSignal));
+      abortSignal.removeEventListener('abort', handleAbort);
+      reject(abortSignal.reason); // eslint-disable-line @typescript-eslint/prefer-promise-reject-errors
```

Comment:
It's the responsibility of the code that calls `abortSignal.abort()` to pass an error (i.e. the suspense fetch store).